Author: George Kiagiadakis <kiagiadakis.george@gmail.com>
Description: Modifies CMakeLists.txt so that libbluedevilaction.so
 is installed in a subdirectory of /usr/lib, as recommended by the
 Debian policy and sets RPATH for everything that uses this library.
 In addition, the command to install actionplugin.h is removed.
Forwarded: not-needed

--- a/src/actionplugins/CMakeLists.txt
+++ b/src/actionplugins/CMakeLists.txt
@@ -6,6 +6,5 @@
 
 target_link_libraries(bluedevilaction ${KDE4_KDEUI_LIBS})
 
-install(TARGETS bluedevilaction ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS bluedevilaction DESTINATION ${LIB_INSTALL_DIR}/bluedevil)
 install(FILES actionplugin.desktop DESTINATION ${SERVICETYPES_INSTALL_DIR})
-install(FILES actionplugin.h DESTINATION include/)
--- a/src/actionplugins/audio/CMakeLists.txt
+++ b/src/actionplugins/audio/CMakeLists.txt
@@ -8,6 +8,7 @@
 kde4_add_plugin(bluedevilaudioactionplugin ${bluedevilaudioactionplugin_SRCS})
 
 target_link_libraries(bluedevilaudioactionplugin ${KDE4_KDECORE_LIBS} ${KDE4_KDEUI_LIBS} ${QT_QTCORE_LIBRARY} ${LibBlueDevil_LIBRARIES} bluedevilaction)
+set_target_properties(bluedevilaudioactionplugin PROPERTIES INSTALL_RPATH ${LIB_INSTALL_DIR}/bluedevil)
 
 install(TARGETS bluedevilaudioactionplugin DESTINATION ${PLUGIN_INSTALL_DIR})
 install(FILES bluedevil-audio.desktop DESTINATION ${SERVICES_INSTALL_DIR})
--- a/src/actionplugins/input/CMakeLists.txt
+++ b/src/actionplugins/input/CMakeLists.txt
@@ -9,6 +9,7 @@
 kde4_add_plugin(bluedevilinputactionplugin ${bluedevilinputactionplugin_SRCS})
 
 target_link_libraries(bluedevilinputactionplugin ${KDE4_KDECORE_LIBS} ${KDE4_KDEUI_LIBS} ${QT_QTCORE_LIBRARY} ${LibBlueDevil_LIBRARIES} bluedevilaction)
+set_target_properties(bluedevilinputactionplugin PROPERTIES INSTALL_RPATH ${LIB_INSTALL_DIR}/bluedevil)
 
 install(TARGETS bluedevilinputactionplugin DESTINATION ${PLUGIN_INSTALL_DIR})
 install(FILES bluedevil-input.desktop DESTINATION ${SERVICES_INSTALL_DIR})
--- a/src/actionplugins/sendfile/CMakeLists.txt
+++ b/src/actionplugins/sendfile/CMakeLists.txt
@@ -12,6 +12,7 @@
 kde4_add_plugin(bluedevilsendfileactionplugin ${bluedevilsendfileactionplugin_SRCS})
 
 target_link_libraries(bluedevilsendfileactionplugin ${KDE4_KDECORE_LIBS} ${KDE4_KDEUI_LIBS} ${QT_QTCORE_LIBRARY} bluedevilaction ${LibBlueDevil_LIBRARIES})
+set_target_properties(bluedevilsendfileactionplugin PROPERTIES INSTALL_RPATH ${LIB_INSTALL_DIR}/bluedevil)
 
 install(TARGETS bluedevilsendfileactionplugin DESTINATION ${PLUGIN_INSTALL_DIR})
 install(FILES sendfile.desktop DESTINATION ${SERVICES_INSTALL_DIR})
--- a/src/actionplugins/audio/helper/CMakeLists.txt
+++ b/src/actionplugins/audio/helper/CMakeLists.txt
@@ -13,6 +13,7 @@
 
 target_link_libraries(bluedevil-audio bluedevilaction
                       ${KDE4_KIO_LIBRARY} ${KDE4_KDEUI_LIBS} ${KDE4_KFILE_LIBS} ${LibBlueDevil_LIBRARIES} bluedevil)
+set_target_properties(bluedevil-audio PROPERTIES INSTALL_RPATH ${LIB_INSTALL_DIR}/bluedevil)
 
 install(TARGETS bluedevil-audio DESTINATION ${INSTALL_TARGETS_DEFAULT_ARGS})
 install(FILES bluedevil-audio.desktop DESTINATION ${XDG_APPS_INSTALL_DIR})
--- a/src/actionplugins/input/helper/CMakeLists.txt
+++ b/src/actionplugins/input/helper/CMakeLists.txt
@@ -14,6 +14,7 @@
 
 target_link_libraries(bluedevil-input bluedevilaction
                       ${KDE4_KIO_LIBRARY} ${KDE4_KDEUI_LIBS} ${KDE4_KFILE_LIBS} ${LibBlueDevil_LIBRARIES} bluedevil)
+set_target_properties(bluedevil-input PROPERTIES INSTALL_RPATH ${LIB_INSTALL_DIR}/bluedevil)
 
 install(TARGETS bluedevil-input DESTINATION ${INSTALL_TARGETS_DEFAULT_ARGS})
 install(FILES bluedevil-input.desktop DESTINATION ${XDG_APPS_INSTALL_DIR})
--- a/src/wizard/CMakeLists.txt
+++ b/src/wizard/CMakeLists.txt
@@ -29,6 +29,7 @@
 
 target_link_libraries(bluedevil-wizard bluedevilaction
                       ${KDE4_KIO_LIBRARY} ${KDE4_KDEUI_LIBS} ${LibBlueDevil_LIBRARIES})
+set_target_properties(bluedevil-wizard PROPERTIES INSTALL_RPATH ${LIB_INSTALL_DIR}/bluedevil)
 
 install(TARGETS bluedevil-wizard ${INSTALL_TARGETS_DEFAULT_ARGS})
 install(FILES bluedevil-wizard.desktop DESTINATION ${XDG_APPS_INSTALL_DIR})
